src/main/resources/avro/reads.avdl

```diff
@@ -177,7 +183,41 @@ record LinearAlignment {
 }
 
 /**
-Each read alignment describes a linear alignment with additional information
+A string-to-reference-graph alignment can be represented by one CIGAR string and
+one `Path` through multiple `Reference`s, against which the CIGAR string is
+interpreted.
+
+Note that `Path`s in `GraphAlignment`s are restricted to visiting `Reference`s
+and following reference adjacencies. If a read needs to be aligned to sequences
+that are not present in a `ReferenceSet`, it needs to be aligned to a new
+`ReferenceSet` with those sequences. If a read needs to follow adjacencies that
+are not present in the `ReferenceSet` it's being aligned to, it should be
+represented as a "chimeric" alignment, and should use multiple `ReadAlignment`s
+and the supplementaryAlignment flag instead of a single `GraphAlignment`.
+
+Some especially large deletions could be represented just as well as a large
+deletion in the CIGAR string, or as a chimeric alignment.
+*/
+record GraphAlignment {
+  /** The `Path` against which the read is aligned */
+  Path path;
+
+  /**
+  The mapping quality of this alignment. Represents how likely
+  the read maps to this position as opposed to other locations.
+  */
+  union { null, int } mappingQuality = null;
+
+  /**
+  Represents the local alignment of this sequence (alignment matches, indels,
+  etc) versus the `Path`.
+  */
+  array<CigarUnit> cigar = [];
```

Comment:
Implementation question: can we require that a cigar operator will be restricted to mapping to a single `Path` in the `Segment`? If I've got a 100bp read that maps perfectly to two 50bp segments, I think that it'd be preferable to have `cigar = { 50=, 50= }` instead of `cigar = { 100=  }`.